### ISSUE TITLE
"docker stop vespa" does not trigger content node shutdown 

### ISSUE THREAD

**Issue by sujitpal** [vespa-engine/vespa#16684]
I was loading a Vespa index using its HTTP Client. My data is in 25 parts and I was loading them in part by part. Current index size is 35 GB and represents 2.8 million records. Last night I ran `docker stop vespa` before machine shutdown, and this morning I restarted the machine and ran `docker start vespa`. I then waited till the ApplicationStatus page returned HTTP 200.

On attempting to run a test query using the search endpoint, I got back this error message in my response.

```
    "errors": [
      {
        "code": 10,
        "summary": "Backend communication error",
        "source": "content",
        "message": "Connection failure on nodes with distribution-keys: 0"
      }
    ]
```

Result of `vespa-get-cluster-state` was as follows:
```
Cluster content is down. Too few nodes available.
content/distributor/0: up
content/storage/0: down
```

I asked about this on the vespa channel on the Relevance Slack workspace, and @jobergum mentioned that this was because the indexes did not sync completely before shutdown, and that the cluster state will continue to show "too few nodes" until the transaction log replays. The `vespa-proton-cmd --local getProtonStatus` (run multiple times) shows that the replay process is working:

```
looking for RTCs matching 'tcp/vespa-container:' (length 20)
found local RTC 'content/search/cluster.content/0/realtimecontroller' with connection spec tcp/vespa-container:19106
"matchengine","CRITICAL","state=OFFLINE","Search interface is offline"
"documentdb:concept","WARNING","state=REPLAY_TRANSACTION_LOG configstate=OK","DocumentDB replay transaction log on startup (22% done)"
```

He also mentioned that there is work going on to make `docker stop vespa` to sync the indexes on shutdown automatically. However, in the meantime, especially for large indexes, he advised to issue `vespa-stop-services` before issuing `docker stop vespa`. Till then, it may make sense to make a note of this in the documentation?

(I ended up stopping the replay and starting back from scratch.)

**Comment by jobergum** [comment]
Thanks for the detailed report, 

The fix for proper docker stop behaviour is in https://github.com/vespa-engine/docker-image/pull/19  which will be released in => 7.365.20 . When this fix is out docker stop will trigger `vespa-stop-services` in the running docker container and this can be used as a work around to stop vespa services properly while we wait for the release of the fix.

 When a proper shutdown is not performed, parts of the unflushed data needs to be replayed from the transaction log as seen from the `DocumentDB replay transaction log on startup (22% done)` and while doing this the node is not considered up and the cluster state is down (as this is a single node setup). 

The default docker stop timeout might be too low for large nodes HNSW index to synch all operations before shutting down as flush will depend on IO capacity so I'll update the documentation on this.

**Comment by jobergum** [comment]
vespaengine/vespa:7.366.33 has a fix for this issue and docker stop will trigger `vespa-stop-services`
